Address feedback on detecting and resolving performance issues (#8833)
mendix/docs: content/en/docs/howto/monitoring-troubleshooting/detect-and-resolve-performance-issues.md

Add microflow timers until you find your culprit microflow, then add extra timers in that microflow to determine which activity is the slow one. When you find a slow activity, see the [Optimizing Microflow Activities](#optimizing) section below. These sections present details on how to optimize your microflow as a whole.

{{% alert color="info" %}}

> Alternatively, you can also use the **TimeMeasureStart** and **TimeMeasureEnd** Java actions provided in the [Community Commons](/appstore/modules/community-commons-function-library/#logging) module. They function similarly as the microflow described in the current section. As added benefits, they can be placed in different microflows without the need to pass the start time from one microflow to another.

Reviewer: ```suggestion
Alternatively, you can also use the **TimeMeasureStart** and **TimeMeasureEnd** Java actions provided in the [Community Commons](/appstore/modules/community-commons-function-library/#logging) module. They function similarly to the microflow described in the current section. As an added benefit, they can be placed in different microflows without the need to pass the start time from one microflow to another.
```

> Alternatively, you can also use the **TimeMeasureStart** and **TimeMeasureEnd** Java actions provided in the [Community Commons](/appstore/modules/community-commons-function-library/#logging) module. They function similarly to the microflow described in the current section. As an added benefit, they can be placed in different microflows without the need to pass the start time from one microflow to another.

Reviewer 2: Can we be more specific here? 
Maybe - 

> They can be placed in any sub-microflow within the same logic flow. 

Or something similar. Basically, we want to make sure that they are called from the same flow. 
Maybe it is also worth mentioning that the `TimeMeasureEnd` will throw an exception if the `TimeMeasureStart` with the same timer name is not present in the flow ??

Oh Btw, I see that there is some issue with the timer when it is called from a different flow. I would not add this statement for now. I'll create a story to investigate and rectify this after talking to the team

Reviewer 2: I discussed with the team. The implementation is fine and you can add this note. 
Please also add a note for the customers saying that an `TimeMeasureEnd` needs to always have `TimeMeasureStart`, else an error will be thrown at runtime.

Author: @akkrishnakumar Thank you so much for the review and for the discussion with the team.

I've made some edits based on what's provided above. I hope that my understanding is correct. Can you please have a look at my commit here: https://github.com/mendix/docs/pull/8833/commits/2e4941154650ecf9b2989bc35efb0d8482d5ce4f?

Reviewer 2: Looks good to me. 👍🏽

Additionally, for details on how denormalization can improve your app performance in some cases, review [How to Denormalize Data to Improve Performance](/howto/data-models/denormalize-data-to-improve-performance/).

{{% alert color="info" %}}

> You can set up the [LogMinDurationQuery](/refguide/custom-settings/#LogMinDurationQuery) for debugging slow database connections or XPaths. For debugging purposes, you can set the value to `500` milliseconds to see all the retrieves or commits that take longer than `500` milliseconds.

Reviewer 3: `LogMinDurationQuery` should be used to debug slow database queries (OQL _and_ XPath), not connections. A slow connection will also likely be logged, but that's not what's being measured here.

Author: @passalaqua Thanks for the comment. Based on this comment, if I change "slow database connections or XPaths" to "slow database queries (OQL and XPaths)" in the info note, would it be correct?

@Michiel-Bijlsma what's your thought here? :)

Reviewer 3: >  if I change "slow database connections or XPaths" to "slow database queries (OQL and XPaths)" in the info note, would it be correct?

Correct!

Author: Commited the change here: https://github.com/mendix/docs/pull/8833/commits/c4c67ac95d376272b376d89bc3fa89fc9c661b62.

> You can set up the [LogMinDurationQuery](/refguide/custom-settings/#LogMinDurationQuery) for debugging slow database queries (OQL and XPath). For debugging purposes, you can set the value to `500` milliseconds to see all the retrieves or commits that take longer than `500` milliseconds.

Reviewer: ```suggestion
You can set the [LogMinDurationQuery](/refguide/custom-settings/#LogMinDurationQuery) custom setting to debug slow database queries (OQL and XPath). For example, for debugging purposes, you could set the value to `500` milliseconds to see all the retrieves or commits that take longer than `500` milliseconds.
```

#### Refresh in Client {#refresh}

> The **Refresh in client** property of a change or commit activity is quite useful to provide updated information to your user. However, when committing large numbers of rows, this can slow you application down, as it attempts to update thousands of rows in your client's browser. Also, a refresh on one object might result in multiple refreshes down the line, for instance, with nested data views. Consider turning it off if possible.

Reviewer: ```suggestion
The **Refresh in client** property of a change or commit activity can provide updated information to your user. However, when committing large numbers of rows, this can slow your application down, as it attempts to update thousands of values in your client's browser. Also, refreshing one object might result in multiple refreshes down the line, for instance, if there are nested data views. Consider turning it off if possible.
```

### Slow Sub-Microflow

If you find that you have a slow sub-microflow, begin the process of identifying the slow activity within the microflow, based on the [Slow Microflows](#slow-micro) section above.

> ### Big Operations and Memory Issue

Author: @Michiel-Bijlsma Does this title make sense?

Reviewer 4: Yes I think the title makes sense :)

> If you notice that with bigger operations, a microflow becomes slower, it might be that there is something kept in memory. For instance, a commit list that grows over 10k objects might affect the application's memory performance.

Reviewer 4: Maybe we can add to this line an extra sentence: To solve these kind of issues consider running the task asynchronous and add batching.

Author: Added in this commit: https://github.com/mendix/docs/pull/8833/commits/bedd45a5a5c20f162ef01e486f2ccc0dfb6c1523.

> If you notice that with bigger operations, a microflow becomes slower, it might be that there is something kept in memory. For instance, a commit list that grows over 10k objects might affect the application's memory performance. To solve such issues, consider running the task asynchronously and adding batching.

Reviewer: ```suggestion
If you notice that with bigger operations, a microflow becomes slower, it might be that there is something kept in memory. For instance, a commit list that grows to over 10k objects might affect the application's memory performance. To solve such issues, consider running the task asynchronously and adding batching.
```
